### PR TITLE
Add UID and GID support for vpc-branch-eni plugin

### PR DESCRIPTION
**Pull request by haikuoliu**
*Issue #, if available:*
https://github.com/aws/amazon-vpc-cni-plugins/issues/32

*Description of changes:*
When both UID and GID are provided, use them to set up tap device
permissions, otherwise look up them using user name.

Tested the change manually

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/vpc-branch-eni/plugin/commands.go [hunk 1]
```diff
@@ -57,14 +58,28 @@ func (plugin *Plugin) Add(args *cniSkel.CmdArgs) error {
...
 
 	// Lookup the user ID for TAP link ownership.
-	var uid int
+	var uid, gid int
 	if netConfig.InterfaceType == config.IfTypeTAP {
-		uid, err = plugin.LookupUser(netConfig.UserName)
-		if err != nil {
-			log.Errorf("Failed to lookup user %s: %v.", netConfig.UserName, err)
-			return err
+		if netConfig.Uid != "" && netConfig.Gid != "" {
```
Comment (ofiliz) [inline plugins/vpc-branch-eni/plugin/commands.go]: What we are doing here is configuration validation, whether UID and GID are valid integers. Therefore this logic should be done in netconfig.go, together with the rest of the validation. Can we please move the strconv calls to netconfig.go line 168?

File: plugins/vpc-branch-eni/config/netconfig.go [hunk 2]
```diff
@@ -37,6 +37,8 @@ type NetConfig struct {
 	BranchGatewayIPAddress net.IP
 	InterfaceType          string
 	UserName               string
+	Uid                    string
```
Comment (ofiliz) [inline plugins/vpc-branch-eni/config/netconfig.go]: I think these should be UID and GID to fit the golang variable naming convention. Also their types should be int, not string. NetConfigJSON contains the fields as they appear in the JSON file. NetConfig contains the parsed fields with correct types after validation. For example, see BranchIPAddress. It is a string in netConfigJSON but net.IPNet in NetConfig.

File: cni/plugin.go [hunk 3]
```diff
@@ -156,8 +156,8 @@ func (plugin *Plugin) printVersionInfo() error {
 	return nil
 }
 
-// LookupUser returns the UID for the given username, or the current user.
-func (plugin *Plugin) LookupUser(userName string) (int, error) {
+// LookupUser returns the UID and GID for the given username, or the current user.
+func (plugin *Plugin) LookupUser(userName string) (int, int, error) {
```
Comment (ofiliz) [inline cni/plugin.go]: This method doesn't really belong in the plugin class. We need to move this logic to another package. Not necessary for this PR, just making a note here.

Comment (haikuoliu) [reply cni/plugin.go]: ok, let's do that separately, let me know if you want me to create an issue for it.

File: plugins/vpc-branch-eni/config/netconfig.go (hunk 2 above)
Comment (haikuoliu) [reply plugins/vpc-branch-eni/config/netconfig.go]: I am following convention here: https://golang.org/pkg/os/user/, if you really like `UID` I can do that as well.

As for type, my concern about using int is we can't really tell if the UID is set or not, as by default the int is set to 0, and you don't know if it's unset or it's set and set to 0. We can set it to -1 to indicate the UID is not set, but to me it's lets preferred as it introduce some implicit assumptions.

File: plugins/vpc-branch-eni/plugin/commands.go (hunk 1 above)
Comment (haikuoliu) [reply plugins/vpc-branch-eni/plugin/commands.go]: Same as previous comment, let me how you think about it then I will try to refactor this if required.

File: plugins/vpc-branch-eni/config/netconfig.go (hunk 2 above)
Comment (ofiliz) [reply plugins/vpc-branch-eni/config/netconfig.go]: Is there any case where UID would not be set when using TAP mode? It seems to me that UID should always be set, hence the value 0 is not ambiguous.

**Comment by haikuoliu** [comment]
@ofiliz I've updated accordingly, ended up remove `UserName` since `UID` and `GID` are now required for TAP mode.

**Pull request merged**